apps/mobile/src/hooks/balances/use-stx-balances.ts

```diff
@@ -0,0 +1,42 @@
+import { AccountStore } from '@/store/accounts/utils';
+import { useStacksSigners } from '@/store/keychains/stacks/stacks-keychains.read';
+import { destructAccountIdentifier } from '@/store/utils';
+
+import { Money } from '@leather.io/models';
+import {
+  useCryptoCurrencyMarketDataMeanAverage,
+  useStxCryptoAssetBalance,
+} from '@leather.io/query';
+import { baseCurrencyAmountInQuote, createMoney, sumMoney } from '@leather.io/utils';
+
+interface StxBalance {
+  stxBalance: Money;
+  stxBalanceUsd: Money;
+}
+
+function GetAccountStxBalance(accountId: string): Money {
```

Comment:
I thought this was a component but indeed think we shouldn't do this, as it's not really how we're supposed to use hooks and will probably cause headaches later.

What's the challenge you're facing here @pete-watters?

Reply:
See comment here: https://github.com/leather-io/mono/pull/542#discussion_r1806573517 

Basically we kind of use hooks for everything and this is proving difficult in this instance where I want to:
- map over the accounts
- fetch a singular account balance using account ID 

I need to use hooks to get account balance - `useStacksSigners` and `useStxCryptoAssetBalance`. 

If I try and do that and use a `hook` for this function AKA `useGetAccountStxBalance`, I get a lint error
```
"useGetAccountStxBalance" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function  react-hooks/rules-of-hooks
```

The functionality works fine but I hit this error. 

By simply changing the name of the function to be `GetAccountStxBalance` it technically becomes a 'React Function component` which returns a `Money` object rather than JSX and that means I can use it inside of `reduce` as I wish

```
function useGetAccountsStxBalance(accounts: AccountStore[]): Money {
  if (!accounts.length) return createMoney(0, 'STX');

  return accounts.reduce(
    (total, account) => sumMoney([total, GetAccountStxBalance(account.id)]),
    createMoney(0, 'STX')
  );
}
```

I am open to doing further work and refactors to try and avoid doing this but I had tried starting down that path and ended up just adding more and more hooks and still having the same issue.

Reply:
Thanks @edgarkhanzadian 👍 I can take a deeper look into more refactors there. 

In the interest of getting some real balance data on screen I was compromising for a fake functional component

Reply:
Yep, exactly. `useQueries` is how you're able to dynamically create an array of individual queries that can then be summed with `combine`

Reply:
Changed this to `useQueries` but more work to follow. See here - https://github.com/leather-io/mono/pull/542#issuecomment-2431713706